### PR TITLE
Added description of the new fucntionality of erase tool

### PR DESCRIPTION
**Pull request by SeqLaz**
This adds a new erase tool for lines and polygons.

https://github.com/opengisch/QField/pull/4584

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-4044 #4584 Brand new geometry editor erase tool functionality](https://app.clickup.com/t/2192114/QF-4044)

File: documentation/how-to/digitize.en.md [hunk 1]
```diff
@@ -139,6 +139,28 @@ A video demonstration of some editing possibilities:
 
 ![type:video](https://player.vimeo.com/video/499564854)<!-- markdown-link-check-disable-line -->
 
+## Erase Tool Functionality
```
Comment (nirvn) [inline documentation/how-to/digitize.en.md:142]: This should be moved to sit alongside the other tool, have it be a 3rd-level header, and harmonize title with the other tools (i.e. Erase tool, not Erase Tool Functionality).

File: documentation/how-to/digitize.en.md [hunk 2]
```diff
@@ -139,6 +139,28 @@ A video demonstration of some editing possibilities:
...
+
+The erase tool is a new feature introduced in the geometry editor, designed to facilitate the removal of parts of a line or polygon geometry. This tool enables users to efficiently correct paths or polygons directly within the application interface and directly from the field.
```
Comment (nirvn) [inline documentation/how-to/digitize.en.md:144]: I'd avoid referring to features as "newly introduced", this doesn't age well.

File: documentation/how-to/digitize.en.md [hunk 3]
```diff
@@ -139,6 +139,28 @@ A video demonstration of some editing possibilities:
...
+
+Suppose you have a polygon representing a building footprint, but there's an unwanted protrusion. You can use the erase tool to precisely remove that portion of the polygon, ensuring accuracy in your geometry representation.
```
Comment (nirvn) [inline documentation/how-to/digitize.en.md:146]: I'm not completely sure I'd call the erase tool "precise" :) The reshape tool is a much more precise tool that'd probably better fit this example.

I'd also note that the erase tool works best with a stylus/pen.

File: documentation/how-to/digitize.en.md [hunk 4]
```diff
@@ -139,6 +139,28 @@ A video demonstration of some editing possibilities:
...
+
+2. **Choose Geometry to Erase**: Click on the line or polygon geometry that you want to edit using the erase tool.
+
+3. **Erase Segments or Portions**:
+
+    - For lines: Click and drag along the segment(s) you wish to erase.
+    - For polygons: Click within the area you want to erase, and drag to define the portion to be removed.
+4. **Confirm Erase Action**: Once you have selected the desired segments or portions to erase, release the mouse button to confirm the erase action.
+
+5. **Finalize Changes**: Save your changes or continue editing as needed.
```
Comment (nirvn) [inline documentation/how-to/digitize.en.md:160]: Wondering if we should get rid of this section. We do not go into that much detailing for other tools. I feel the two paragraphs introducing the feature + a screencast should be enough. We already go into more details on how to open the geometry editor toolbox to begin with.

File: documentation/how-to/digitize.en.md (hunk 1 above)
Comment (SeqLaz) [reply documentation/how-to/digitize.en.md:142]: What do you think if it moved to the same section of reshape tool?

File: documentation/how-to/digitize.en.md (hunk 4 above)
Comment (SeqLaz) [reply documentation/how-to/digitize.en.md:160]: Yes, I think is better!

File: documentation/how-to/digitize.en.md (hunk 3 above)
Comment (SeqLaz) [reply documentation/how-to/digitize.en.md:146]: Do you think is better to replace the gif, and also add this quote "for more precision utilize with a stylus/pen"?

File: documentation/how-to/digitize.en.md [hunk 5]
```diff
@@ -128,6 +128,14 @@ The reshape tool allows you to change line and polygon geometries by drawing sha
 result in a reshaped line following the edge of the drawn shape or a polygon with the drawn shaped
 used to clip or expand the geometry.
 
+## Reshape eraser tool
```
Comment (nirvn) [inline documentation/how-to/digitize.en.md:131]: This needs to be a 3rd level header (like the other tools); you'll also need to add the tool in the list of available tools at the beginning of the section.

File: documentation/how-to/digitize.en.md [hunk 6]
```diff
@@ -128,6 +128,14 @@ The reshape tool allows you to change line and polygon geometries by drawing sha
...
+
+The reshape eraser tool is a feature in the geometry editor, designed to facilitate the removal of parts of a line or polygon geometry. This tool enables to efficiently correct paths or polygons directly within the application interface and directly from the field conditions.
+
+Suppose you have a polygon representing a building footprint, but there's an unwanted protrusion. You can use the erase tool to precisely remove that portion of the polygon, ensuring accuracy in your geometry representation.
```
Comment (nirvn) [inline documentation/how-to/digitize.en.md:135]: I'd remove that line entirely, just stick to a on paragraph description + the webm.

File: documentation/how-to/digitize.en.md [hunk 7]
```diff
@@ -128,6 +128,14 @@ The reshape tool allows you to change line and polygon geometries by drawing sha
 result in a reshaped line following the edge of the drawn shape or a polygon with the drawn shaped
 used to clip or expand the geometry.
 
+## Reshape eraser tool
+
+The reshape eraser tool is a feature in the geometry editor, designed to facilitate the removal of parts of a line or polygon geometry. This tool enables to efficiently correct paths or polygons directly within the application interface and directly from the field conditions.
```
Comment (nirvn) [inline documentation/how-to/digitize.en.md:133]:
```suggestion
The reshape eraser tool is designed to ease the removal of parts of a line or polygon geometry. The tool mimics eraser tools from 2D drawing programs and works best with a stylus.
```

**Comment by m-kuhn** [comment]
@SeqLaz what do you think about the suggestions by @nirvn ?

File: documentation/how-to/digitize.en.md (hunk 6 above)
Comment (SeqLaz) [reply documentation/how-to/digitize.en.md:135]: Sorry for the delayed response. Thank you, @nirvn

File: documentation/how-to/digitize.en.md (hunk 5 above)
Comment (SeqLaz) [reply documentation/how-to/digitize.en.md:131]: Great, thank you @nirvn!

**Review by nirvn — APPROVED** [review]

**Comment by nirvn** [comment]
We can merge this.

**Pull request merged**